### PR TITLE
log-filter will now match multiple lines

### PR DESCRIPTION
**Pull request by brototyp**
Before this, when one selected the "Error"-Filter or such, it would only match the first line of this log message. Now it will select the whole message. The same is true for searching.

### CODE REVIEW

File: Classes/TBODeveloperOverlayLogger/TBODeveloperOverlayFilterableLogFormatterProtocol.h [hunk 1]
```diff
@@ -34,6 +34,16 @@
...
 
+@optional
+
+/**
+ *  Use this method to define what is a single log message. This way you can have one log message span multiple lines.
+ *  If not defined, one message is expected to be one line.
+ *
+ *  @return the NSRegularExpression that matches exactly one log message.
+ */
+- (NSRegularExpression *_Nullable)logMessageRegularExpression;
```
Comment (brototyp) [inline Classes/TBODeveloperOverlayLogger/TBODeveloperOverlayFilterableLogFormatterProtocol.h:46]: This is the magic here. Basically a log formatter now can define where a log message starts and where it ends. This is done with a regEx that always matches a full log message.

**Comment by brototyp** [comment]
One thing:

The question is if -logMessageRegularExpression: matches including a newline or without. The current implementation expects it to match including the new line. That is the reason why I append a newline to the log in TBODeveloperOverlayLoggerCocoaLumberjackDatasource:90. That is definitely not a beautiful solution.

An alternative would be to expect that there are never two log messages in one line. Thus the regex doesn't include the newline. Then we could join all matches via a newline in TBODeveloperOverlayLoggerCocoaLumberjackDatasource-filteredLog:

Is there a case where one would want to have multiple log-messages in a single line?

**Comment by brototyp** [comment]
Ok, I went with the logMessageRegularExpression to not contain the newline.

File: Classes/TBODeveloperOverlayLogger/NSString+RegexMatches.h [hunk 2]
```diff
@@ -12,22 +12,7 @@
...
-
-/**
- * Returns a substring if any of the given regular expressions matches.
- *
- *  @param regexes A NSArray of NSRegularExpressions to search for matching substrings.
- *
- *  @return The substring that matches any of the regexes; If there are more than one match, all matches will be concatenated with a newline.
- */
-- (NSString *_Nonnull)substringThatMatchesAnyRegex:(NSArray <NSRegularExpression *> *_Nonnull)regexes;
+- (BOOL)matchesRegex:(NSRegularExpression *)regex;
```
Comment (thorstenstark) [inline Classes/TBODeveloperOverlayLogger/NSString+RegexMatches.h:23]: What about documentation here?

File: Example/AppDelegate.m [hunk 3]
```diff
@@ -31,6 +31,8 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     DDLogWarn(@"DidFinishLaunchingWithOptions:");
     DDLogInfo(@"DidFinishLaunchingWithOptions:");
     
+    DDLogVerbose(@"Multiline\nLog\nMessage");
```
Comment (thorstenstark) [inline Example/AppDelegate.m:34]: What is this for?

File: TBODeveloperOverlayTests/NSString+RegexMatchesTests.m [hunk 4]
```diff
@@ -1,76 +0,0 @@
-//
```
Comment (thorstenstark) [inline TBODeveloperOverlayTests/NSString+RegexMatchesTests.m:1]: Why did you remove the tests?

File: Example/AppDelegate.m (hunk 3 above)
Comment (brototyp) [reply Example/AppDelegate.m:34]: So we have a Multiline Log message in the example app.

Comment (thorstenstark) [reply Example/AppDelegate.m:34]: Oh yes, it's the example here... all right.

File: TBODeveloperOverlayTests/NSString+RegexMatchesTests.m (hunk 4 above)
Comment (brototyp) [reply TBODeveloperOverlayTests/NSString+RegexMatchesTests.m:1]: Because I deleted the methods they tested.

File: Classes/TBODeveloperOverlayLogger/NSString+RegexMatches.h (hunk 2 above)
Comment (brototyp) [reply Classes/TBODeveloperOverlayLogger/NSString+RegexMatches.h:23]: Good Point. I forgot those, because the method is quite self explanatory. But still. We should add some.

File: TBODeveloperOverlayTests/NSString+RegexMatchesTests.m (hunk 4 above)
Comment (thorstenstark) [reply TBODeveloperOverlayTests/NSString+RegexMatchesTests.m:1]: But in the same file you added two new methods, that are not tested, as far as i can see...
